### DISCUSSION TITLE
Linear Interpolation Between AMR Levels

### DISCUSSION THREAD

**Discussion by Abisleiman** [AMReX-Codes/amrex#3056] [Q&A]
Does linear interpolation between AMR levels exist as a feature in AMReX? When a cell is refined it uses a piecewise constant interpolation to the new level currently. Is there a way to use a linear interpolation to the newly created level upon refinement. Piecewise constant interpolation is a limiting factor in increasing AMR levels where differing gradients are used to tag or when a large increase in AMR level is desired. In this situation artificially steep gradients are created that lead to stability issues.

**Comment by WeiqunZhang** [comment]
> When a cell is refined it uses a piecewise constant interpolation to the new level currently.

Not sure what you mean.  AMReX provides a number of interpolaters.  It's up to the user which one to use.  For example, these two tests both use `cell_cons_interp`, which is linear interpolation with limiting.  (https://github.com/AMReX-Codes/amrex/blob/development/Src/AmrCore/AMReX_Interpolater.cpp#L37)

 - https://github.com/AMReX-Codes/amrex/blob/development/Tests/Amr/Advection_AmrCore/Source/AmrCoreAdv.cpp#L457

 - https://github.com/AMReX-Codes/amrex/blob/development/Tests/Amr/Advection_AmrLevel/Source/AmrLevelAdv.cpp#L132

**Reply by ZiyangHuang** [reply]
Hi Weiqun,

I check the link you gave (https://github.com/AMReX-Codes/amrex/blob/development/Src/AmrCore/AMReX_Interpolater.cpp#L37), and saw that ```CellConservativeLinear    lincc_interp;``` and ```CellConservativeLinear    cell_cons_interp(false);```. It seems that ```lincc_interp``` is the one with limiting and ```cell_cons_interp``` is the one without. Moreover, what limter is applied in ```CellConservativeLinear```, and is the limiter TVD? 

Thanks!

**Reply by WeiqunZhang** [reply]
Both `lincc_interp` and `cell_cons_interp` do limiting. The difference is their approach for multiple variables. For `cell_cons_interp`, each variable is handled independently, whereas for `lincc_interp` all variables are limited by the same factor (that is their second-order centered difference slopes are reduced by the same factor). The limiter used in `CellConservativeLinear` is the the monotonized central limiter, which is second-order TVD.
